internal/commands/pull.go

```diff
@@ -77,6 +100,57 @@ func NewPullCommand(ctx context.Context) *cobra.Command {
 	}
 
 	cmd.Flags().StringP("policy", "p", "policy", "Path to download the policies to")
+	cmd.Flags().String("verify", "none", "Verify the image with none|cosign. Default none")
+	cmd.Flags().String("cosign-key", "", "path to the public key file, KMS, URI or Kubernetes Secret")
 
 	return &cmd
 }
+
+func verifyCosign(ctx context.Context, rawRef string, keyRef string) error {
+
+	digest, err := crane.Digest(rawRef)
+
+	if err != nil {
+		return fmt.Errorf("unable to resolve digest for an image %s: %v\n", rawRef, err)
+	}
+
+	rawRef = rawRef + "@" + digest
+
+	log.Printf("verifying image: %s\n", rawRef)
+
+	cosignExecutable, err := exec.LookPath("cosign")
```

Comment:
although the issue is closed, I wanted to raise a point: would it be possible making this via lib?

Reply:
oh seems it has already been discussed, is there any plan to use signing/verifying capabilities as external lib? or an issue that can be linked also here?